src/Commands/Domain/SetPrimaryCommand.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+
+namespace Pantheon\Terminus\Commands\Domain;
+
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Commands\WorkflowProcessingTrait;
+use Pantheon\Terminus\Models\Domain;
+use Pantheon\Terminus\Models\Environment;
+use Pantheon\Terminus\Models\Site;
+use Pantheon\Terminus\Models\Workflow;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * Class SetPrimaryCommand
+ * @package Pantheon\Terminus\Commands\Domain
+ */
+class SetPrimaryCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    use WorkflowProcessingTrait;
+
+    /**
+     * Sets the primary domain for a site and environment, causing all traffic to redirect to the primary domain.
+     *
+     * @authorize
+     *
+     * @command domain:primary:set
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     * @param string $domain A domain that has been associated to your site. Optional when running interactively.
+     *
+     * @usage domain:primary:set <site_env>
+     */
+    public function set($site_env, $domain)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        // The primary domain is set via a workflow so as to use workflow logging to track changes.
+        $this->log()->notice('Setting primary domain to {domain}...', ['domain' => $domain]);
+        $this->runWorkflowForEnvironmentAndDomain($env, $domain);
+    }
+
+    /**
+     * Unsets the primary domain for a site and environment.
+     *
+     * @authorize
+     *
+     * @command domain:primary:unset
```

Comment:
Maybe `domain:primary:clear`?

Take a straw poll. 😄

Reply:
I realized that we're already calling this "reset"ting the primary domain in platform code and documentation; my calling it anything else is just a potential source of confusion. I've updated it to `domain:primary:reset` for consistency. How's that sound?

Reply:
This is a product decision, ultimately. @ari-gold

Reply:
("reset" sounds good to me)

Reply:
Let's go with "set" / "unset" ... I think "reset" is ambiguous and "unset" is more clear.

Reply:
Thanks for the explanation @TeslaDethray.  I'm +1 on add/remove

P.S. It was helpful to search for various keywords (set, unset, add, remove, create, delete, enable, disable) on https://pantheon.io/docs/terminus/commands to see what our current conventions are.